packages/cli/src/number-parse.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Parses the input and returns it as a number if it's valid, otherwise throws error.
+ *
+ * @param {string | number} input
+ * @returns {number | undefined}
+ */
+export const parseNumber = input => {
+  if (typeof input === 'number' && !Number.isNaN(input)) {
+    return input;
+  }
+
+  if (typeof input === 'string' && /^-?\d+(\.\d+)?$/.test(input.trim())) {
+    return Number(input);
+  }
```

Comment:
A thing you can rely on is that `Number(string)` will return `NaN` if the string doesn’t parse as any valid JavaScript representation of a `number`.

You can also assume `parseNumber` receives `string` and will never see a `number` as input, given that this is not a general purpose utility but specifically for parsing numbers for the CLI.

Reply:
(Notably, this will expand the range of expressible numbers to include `1e3` without requiring any additional complexity in a regex.)

Reply:
An alternative suggestion that I especially like for CLIs, from #2204: https://github.com/endojs/endo/blob/20b4d24ae9db95dfb8073090ca0ff572868ea5a0/scripts/esbench.mjs#L110

example valid inputs:
* "42"
* "-40.0"
* ".5"
* "1.337e3"
* "␣+1␣" (_where "␣" is a visible representation of whitespace_)
* "0B111" (_binary for `7`_)
* "0o040" (_octal for `32`_)
* "0xf00" (_hexadecimal for `3840`_)

example invalid inputs (_i.e., they return `NaN`_):
* "f00"
* "NaN"
* "Infinity"
* "7up"
* "␣"
* (empty string)

Reply:
Thanks! I also added more test cases based on @gibson042 examples. @kriskowal What is weird is that `Number('')` returns `0` instead of `NaN`